src/bin/cargo/commands/package.rs

```diff
@@ -1,18 +1,21 @@
 use crate::command_prelude::*;
 
-use cargo::ops::{self, PackageOpts};
+use cargo::ops;
+use cargo::ops::PackageListFormat;
+use cargo::ops::PackageOpts;
 
 pub fn cli() -> Command {
     subcommand("package")
         .about("Assemble the local package into a distributable tarball")
         .arg_index("Registry index URL to prepare the package for (unstable)")
         .arg_registry("Registry to prepare the package for (unstable)")
         .arg(
-            flag(
+            optional_opt(
                 "list",
-                "Print files included in a package without making one",
+                "Print files included in a package without making one: (unstable) json",
             )
-            .short('l'),
+            .short('l')
+            .value_name("FMT"),
```

Comment:
Should we instead use `--message-format`?

One reason is I wonder if we should focus on ndjson messages rather than a single json blob.  This makes it easier to extend with more feature.  For example, we can't extend `cargo metadata` with warnings or errors without adding them to that single schema which feels weird.

Reply:
I've taken `message` to mean "output message" which is generic, not specific to the type of message.  The shape would probably need to be consistent across the different messages though.

Reply:
In the future `--message-format=json` would also affect status message Cargo emits, right? So here comes another counterpoint: This adds some more work on users side, if they want to keep the human-readable status messages.

Reply:
> In the future `--message-format=json` would also affect status message Cargo emits, right? So here comes another counterpoint: This adds some more work on users side, if they want to keep the human-readable status messages.

Thats true when interacting with most of our json APIs and why some have asked for a way to tee.